### ISSUE TITLE
Loads successfully but isn't working from java button or wake word

### ISSUE THREAD

**Issue by Shagans982** [alexa/alexa-avs-sample-app#750]
I have all 3 terminals running and they seem to be successfully running yet it doesn't seem to pick up my voice.  I am using a standard USB mic, same as listed in the items list. I've tried 2 different ones, in multiple USB slots. Any suggestions? When I run **arecord -l** I get this.

        pi@raspberrypi:~ $ arecord -l
        **** List of CAPTURE Hardware Devices ****
        card 1: SoundBar [Dell AC511 USB SoundBar], device 0: USB Audio [USB Audio]
        Subdevices: 1/1
        Subdevice #0: subdevice #0

Terminal 1
       
        pi@raspberrypi:~/Desktop/alexa-avs-sample-app/samples $ cd companionService && npm start

        > alexa-voice-service-sample-companion-service@1.0.0 start /home/pi/Desktop/alexa-avs-       sample-app/samples/companionService
         > node ./bin/www

        This node service needs to be running to store token information memory and vend them for the    AVS app.

         Listening on port 3000
        Successfully retrieved access token for session id: xxxxxxxxxxf5844f21612
        Successfully retrieved access token for session id: xxxxxxxxxxf5844f21612
        Successfully retrieved access token for session id: xxxxxxxxxxf5844f21612
        Successfully retrieved access token for session id: xxxxxxxxxxf5844f21612

Terminal 2
        
        pi@raspberrypi:~/Desktop/alexa-avs-sample-app/samples $ cd javaclient && mvn exec:exec
        [INFO] Scanning for projects...
        [INFO]                                                                         
        [INFO] ------------------------------------------------------------------------
        [INFO] Building Alexa Voice Service Sample Java Client 20160207.3
        [INFO] ------------------------------------------------------------------------
        [INFO] 
        [INFO] --- exec-maven-plugin:1.2.1:exec (default-cli) @ sample-java-client ---

Terminal 3

       pi@raspberrypi:~/Desktop/alexa-avs-sample-app/samples $ cd wakeWordAgent/src &&       ./wakeWordAgent -e sensory
       INFO:main: Starting Wake Word Agent
       INFO:WakeWordAgent: State set to IDLE(2)
       INFO:Initializing Sensory library | library name: TrulyHandsfree | library version: 5.0.0-avs.4 | model     file: ../ext/resources/spot-alexa-rpi.snsr
        WARNING:Library expires on: License expires on 28 Aug 2017 00:00:00 GMT
        INFO:SensoryWakeWordEngine: mainLoop thread started
        INFO:WakeWordAgent: thread started
        ERROR:An error happened in the mainLoop of SensoryWakeWord snsrRun(): task: Input channel audio-pcm: ALSA error: No such file or directory
        INFO:SensoryWakeWordEngine: mainLoop thread ended
        INFO:WakeWordIPCSocket::mainLoop thread started
       INFO:WakeWordIPCSocket: init socket on port:5123
       INFO:===> Connected to AVS client <===

**Comment by elstryan** [comment]
Hi, 
Everything in those windows appears correct to me. What happens when you try pushing the microphone button in the java app? Does the 'Listening' text appear? This should happen after you press that button.

Additionally can you share the application.log after pressing the microphone button? You can find this under samples/javaclient/logs then we can see if we can spot anything that looks off.

**Comment by Shagans982** [comment]
Yep, the button fires and says listening and I hear the beep through my speaker. I'll post the logs shortly

**Comment by Shagans982** [comment]
Correct me if I am wrong but I am supposed to see my USB mic under devices for record -l and speaker under aplay -l ? I am seeing the flip of that. Mic under play and speaker under record.

**Comment by sidfrgaming** [comment]
Please help me i have the same issue 
Neither the gui button nor the wake word agent work 
When i click on the mic button an error dialog box pop-up appears

**Comment by sidfrgaming** [comment]
Please help me i have the same issue 
Neither the gui button nor the wake word agent work 
When i click on the mic button an error dialog box pop-up appears

**Comment by yugoren** [comment]
@Shagans982: Could you try running `arecord -d 2 test.wav` and replay it with `aplay test.wav` to see if the channels are working properly?

@sidfrgaming: Could you tell us what the error dialog says?

**Comment by mhaisekar** [comment]
I have the same problem, no error across all three windows, but no response. When I press the mic button, it acts as if it is listening something but there is no response back. Looks like the app is working fine from a listening point of view, but it does not transfer the 'voice question' to amazon for it to register and respond back. Something seems to be broken there.

**Comment by mhaisekar** [comment]
btw, I tried to run arecord -d 2 test.wav
and it errored out saying Device or resource busy

**Comment by dskidmor** [comment]
Hey @mhaisekar, I could use some more information to help troubleshoot:

- Can you confirm what kind of microphone you are using? For example, a USB mic.
- What is the output of `arecord -l`? This should list audio recording devices on your system.
- Could you share the contents your [`.asoundrc`](http://alsa.opensrc.org/Asoundrc#Where_does_.asoundrc_live.3F) file?

**Comment by eric-200** [comment]
Hi together,

I have the same problem and no solution. If I write the command per hand.
 (cd alexa-avs-sample-app/samples/wakeWordAgent/src
./wakeWordAgent -e sensory)
all fine but not in a .sh script.

arecord -l
`**** List of CAPTURE Hardware Devices ****
card 1: C725 [Plantronics C725], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
`
.asoundrc

`pcm.!default {
  type asym
   playback.pcm {
     type plug
     slave.pcm "hw:0,0"
   }
 capture.pcm {
   type plug
    slave.pcm "hw:1,0"
   }
}
`



sorry for my bad englisch ;) I hope somebody can help. I search so much but I didn’t find a solution for me.

**Comment by theMooseChen** [comment]
Hi
Did you try running `arecord -d 2 test.wav` and replay it with `aplay test.wav` to see if the channels are working properly?

Additionally can you share the application.log after pressing the microphone button? You can find this under samples/javaclient/logs then we can see if it is the microphone problem or the app problem. Thanks!

**Comment by mhaisekar** [comment]
the microphone that I was using was crappy. I replaced the microphone and it worked fine. It was more of a hardware issue and it was not recording properly.

**Comment by theMooseChen** [comment]
Thanks for your followup and glad you solved your problem! @mhaisekar

**Comment by eric-200** [comment]
i am not sure why, but, now all fine.

alexa-start.sh

`#!/bin/sh

cd /home/pi/alexa-avs-sample-app/samples

cd companionService && npm start &

sleep 5s

cd /home/pi/alexa-avs-sample-app/samples
cd javaclient && mvn exec:exec &

sleep 30s

cd /home/pi/alexa-avs-sample-app/samples/wakeWordAgent/src &&  ./wakeWordAgent -e sensory &
`
Thank you anyway.
